crates/trigger/src/cli.rs

```diff
@@ -35,7 +39,7 @@ where
             short = 'L',
             long = "log-dir",
             )]
-    pub log: Option<PathBuf>,
+    pub log: Option<Option<PathBuf>>,
```

Comment:
If we are going with this UI, we should at least rename this option.  `--log-dir` with no value doesn't connote "log to the default place."  Seems like this also changes the behaviour to not write a log file by default, which doesn't seem to fit what Mikkel was asking for in the original issue.

Reply:
I was thinking about renaming it, but I couldn't find any option better than this.
When the default output is stdout, this option means to me `log to directory`, and no value corresponds the default location.

If it's made explicit in the help output like this:

```
 -L, --log-dir <APP_LOG_DIR>
            Log directory for the stdout and stderr of components [default: $HOME/.spin/<app-name>/logs]
```

it should be an acceptable solution (at least for now) with an ability to re-iterate it later.

Another option is to introduce separate argument, and use both together like `--log-destination file --log-dir <..>` but often this ends up with dozens of different options and confuses the user.

Regarding not writing to the log file by default:
- the fact that spin is implicitly writing logs somewhere was quite surprising to me. I couldn't find anything about it in the docs or help output (there is a chance that I missed it);
- another issue is that that at this moment there is no way to disable logging into the file which might be an issue in some restricted environments (https://github.com/fermyon/spin/issues/815);
- currently, with `--follow-all` options spin writes to both files and *stderr*, what surprised me until I looked into the source code (https://github.com/fermyon/spin/blob/main/crates/trigger/src/stdio.rs#L39).
When I ran spin like:

```
$ spin build --up --follow-all 2> /tmp/error.log
```

`println!` output (from the spin app) also appeared in the error log.

What I like about writing to std{out|err} by default is that this gives the flexibility to consume logs the way you like it:
- redirect into the file;
- put everything into the `/dev/null` because YOLO;
- redirect it into the log collector of your choice;